src/components/lists/CharaList/CharaList_CharaUpgradeButtons.jsx

```diff
@@ -18,28 +18,47 @@ import notteWtfIcon from '../../../assets/icons/nottewtf.png';
 
 const MAX_AUGMENT_COUNT = 100;
 
-function CharaList_CharaUpgradeButtons({adventurerId, adventurerMeta}) { 
-  
+function CharaList_CharaUpgradeButtons({adventurerId, adventurerMeta}) {
+
   const dispatch = useDispatch();
-  const { addAdventurerStory, maxAdventurer } = useDragaliaActions();
+  const { addAdventurerStory, removeAdventurerStory, maxAdventurer } = useDragaliaActions();
 
   const adventurerObject = useSelector(state => state.jsonData.data.chara_list
     .find(adventurerObject => adventurerObject["chara_id"] === adventurerId));
-  
+
   const isOwned = adventurerObject ? true : false;
-    
+
   const onGet = () => {
     const newAdventurerObject = DragaliaUtils.getNewAdventurer(adventurerMeta);
     dispatch(addJsonDataListObject("chara_list", newAdventurerObject));
     addAdventurerStory(adventurerId, true);
     dispatch(addToObjectListObjectField(
-      "fort_bonus_list", "chara_bonus_by_album", "elemental_type", 
+      "fort_bonus_list", "chara_bonus_by_album", "elemental_type",
       adventurerMeta.ElementalTypeId, "hp", 0.1));
     dispatch(addToObjectListObjectField(
-      "fort_bonus_list", "chara_bonus_by_album", "elemental_type", 
+      "fort_bonus_list", "chara_bonus_by_album", "elemental_type",
       adventurerMeta.ElementalTypeId, "attack", 0.1));
   }
 
+  const onRemove = () => {
+    const MEGA_MAN = 10750102;
+    const PRINCE = 10140101;
+
+    if (adventurerId === MEGA_MAN || adventurerId === PRINCE) {
+      return;
+    }
+
+    dispatch(removeJsonDataListObject("chara_list", "chara_id", adventurerId));
+    removeAdventurerStory(adventurerId);
+
+    dispatch(addToObjectListObjectField(
```

Comment:
This should handle the case where the adventurer to be removed is upgraded, in which case it may have contributed more than +0.1% to the encyclopedia bonuses. According to the [wiki](https://dragalialost.wiki/w/Notte%27s_Notes#Stat_Bonuses) an adventurer can contribute up to +0.3% HP and Strength to the encyclopedia bonuses, based on their level and mana circle unlock count. Logic similar to this can be found in `DragaliaActionsUtils.js::maxAdventurer and handleDragonEncyclopedia`, which takes in an existing adventurer or dragon object and calculates the amount of bonus to add based on the upgrade progress. Perhaps a util function like `removeAdventurer` can be refactored to `DragaliaActionsUtils.js` to handle all the logic for removing an adventurer (handle remove from `chara_list`, handle story removal, handle encyclopedia bonus undo)